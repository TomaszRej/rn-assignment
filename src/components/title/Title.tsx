import React, {FC} from 'react';
import {StyleSheet, Text, TextProps} from 'react-native';
import {Colors} from '../../constants';

const Title: FC<TextProps> = props => {
  return (
    <Text style={styles.title} {...props}>
      {props.children}
    </Text>
  );
};

export default Title;

const styles = StyleSheet.create({
  title: {
    fontSize: 32,
    marginBottom: 20,
    color: Colors.text,
  },
});
