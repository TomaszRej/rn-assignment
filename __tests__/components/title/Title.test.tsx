import React from 'react';
import {render, RenderAPI} from '@testing-library/react-native';
import Title from '../../../src/components/title/Title';

describe('Title', () => {
  let component: RenderAPI;
  const text = 'testText';

  beforeEach(() => {
    component = render(<Title>{text}</Title>);
  });

  describe('Title snapshot', () => {
    it('should render correct snapshot for archivedListItem', () => {
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Title props', () => {
    it('should render with correct text', () => {
      const {getByText} = component;
      expect(getByText(text)).toBeTruthy();
    });
  });
});
