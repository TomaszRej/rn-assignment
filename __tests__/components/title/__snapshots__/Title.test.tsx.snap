// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Title snapshot should render correct snapshot for archivedListItem 1`] = `
<Text
  style={
    Object {
      "color": "black",
      "fontSize": 32,
      "marginBottom": 20,
    }
  }
>
  testText
</Text>
`;
